kernel/typeops.ml

```diff
@@ -499,6 +499,10 @@ let type_of_case env (mib, mip) ci u pms (pctx, p, pt) iv c ct _lf lft =
   let ci = if ci.ci_relevance == rp then ci
     else (warn_bad_relevance_ci (); {ci with ci_relevance=rp})
   in
+  let () = match ci.ci_relevance with
+  | Sorts.Relevant | Sorts.Irrelevant -> ()
+  | Sorts.RelevanceVar _ -> anomaly Pp.(str "the kernel does not support sort variables")
+  in
```

Comment:
```suggestion
  let () = match ci.ci_relevance with
  | Sorts.Relevant | Sorts.Irrelevant -> ()
  | Sorts.RelevanceVar _ -> anomaly Pp.(str "the kernel does not support sort variables")
  in
  let ci = if ci.ci_relevance == rp then ci
    else (warn_bad_relevance_ci (); {ci with ci_relevance=rp})
  in
```
otherwise the error can never trigger since ci_relevance has been fixed to == rp which comes from an already-typed value.

Reply:
This code is gonna die soon, I have a branch that removes this hacks anyways. Since it's there only just in case, I'd just leave it there.

Reply:
as you wish